src/components/common/ColonyActions/ActionDetailsPage/ActionDetailsPageFeed/ActionDetailsPageFeed.tsx

```diff
@@ -10,21 +9,18 @@ const displayName =
 
 interface ActionsPageFeedProps {
   actionData: ColonyAction;
-  // networkEvents: FormattedEvent[];
 }
 
-const ActionDetailsPageFeed = ({
-  actionData,
-}: // networkEvents,
-ActionsPageFeedProps) => {
-  const events = ACTIONS_EVENTS[actionData.type];
+const ActionDetailsPageFeed = ({ actionData }: ActionsPageFeedProps) => {
+  const motionEvents = actionData.motionData?.events;
   return (
     <>
-      {events?.map((event) => (
+      {motionEvents?.map((event) => (
         <ActionsPageEvent
           actionData={actionData}
-          eventName={event}
-          key={event}
+          motionEventData={event}
+          eventName={ColonyAndExtensionsEvents[event.name]}
+          key={`${event.name}-${event.initiatorAddress}-${event.amount}`}
```

Comment:
This key is temporary until a better key can be created when more of the messages are implemented.

Reply:
This is a really bad key, and you won't be able to solve with more values, and it will only get worse with time as we surface more and more events in that feed.

The only way to properly do this, is to also store the `transactionHash` of the transaction that emmited this event and the `logIndex` of the specific event.

Then do `${transactionHash}${logIndex}` which will give you a truly unique key.

Reply:
Seconding my comments in the block ingestor, we can use the array index here. It would certainly be simpler and is not going to affect how [react renders the list](https://react.dev/learn/rendering-lists) because the list ordering does not change.

Reply:
The key is now created in the block ingestor as messageKey